src/popper/translators/translator_task.py

```diff
@@ -13,11 +13,36 @@ def __init__(self):
         super().__init__()
 
     def translate(self, wf):
-        box = Box(version="3", tasks={}, vars={"PWD": {"sh": "pwd"}})
+        box = Box(version="3", tasks={}, vars={})
+
+        box["vars"] = {
+            "PWD": {"sh": "pwd"},
+            # redirect stderr to /dev/null to supress warning if the directory is not a git repo
+            "GIT_COMMIT": {"sh": 'git rev-parse HEAD || echo ""'},
+            "GIT_BRANCH": {"sh": 'git branch --show-current 2>/dev/null || echo ""'},
+            "GIT_SHA_SHORT": {
+                "sh": 'git rev-parse --short HEAD 2>/dev/null || echo ""'
+            },
+            "GIT_REMOTE_ORIGIN_URL": {
+                # git config --get exists with non-zero code if the config is not set.
+                # if the remote origin url is not set, set the empty string
+                # no need for redirect since it doesn't generate an error message
+                "sh": 'git config --get remote.origin.url || echo ""'
+            },
+            "GIT_TAG": {"sh": "git tag -l --contains HEAD 2>/dev/null | head -n 1"},
+        }
+
+        box["env"] = {
+            "GIT_COMMIT": "{{.GIT_COMMIT}}",
+            "GIT_BRANCH": "{{.GIT_BRANCH}}",
+            "GIT_SHA_SHORT": "{{.GIT_SHA_SHORT}}",
+            "GIT_REMOTE_ORIGIN_URL": "{{.GIT_REMOTE_ORIGIN_URL}}",
+            "GIT_TAG": "{{.GIT_TAG}}",
+        }
 
         # translate each step
         for step in wf["steps"]:
-            box["tasks"][step["id"]] = self._translate_step(step)
+            box["tasks"][step["id"]] = self._translate_step(step, box["env"])
```

Comment:
pass `box["env"]` which is needed by `translate_docker_step` to create a list of environment variable